falcon/api.py

```diff
@@ -544,13 +545,13 @@ def _call_req_mw(self, stack, req, resp):
             # Put executed component on the stack
             stack.append(component)  # keep track from outside
 
-    def _call_rsrc_mw(self, stack, req, resp, resource):
+    def _call_rsrc_mw(self, stack, req, resp, resource, params):
         """Run process_resource middleware methods."""
 
         for component in self._middleware:
             _, process_resource, _ = component
             if process_resource is not None:
-                process_resource(req, resp, resource)
+                process_resource(req, resp, resource, params)
```

Comment:
Are we sure we want to take this path?
In my opinion, 0.x versions are made to keep maintainers out from 100% code backward compatibility. And specifically is this case we are only talking about adding a parameter to "process_resource" middlewares methods. It does not affect any logic or API behaviour. So instead, I would just add this breaking change in the release note, and let falcon users *just* go through every middleware and add an argument.
In other words, I think the balance here, in this specific case, is in favour of adding a bit of charge to falcon users instead of adding more code complexity and legacy on the shoulders of the maintainer.
Thoughts?